hawkular-command-gateway-war/src/main/java/org/hawkular/cmdgw/ws/server/FeedListenerGenerator.java

```diff
@@ -1,165 +0,0 @@
- * See the License for the specific language governing permissions and
- * limitations under the License.
- */
-package org.hawkular.cmdgw.ws.server;
-
-import java.util.ArrayList;
-import java.util.HashMap;
-import java.util.List;
-import java.util.Map;
-
-import javax.annotation.PostConstruct;
-import javax.annotation.Resource;
-import javax.ejb.EJB;
-import javax.ejb.Singleton;
-import javax.ejb.Startup;
-import javax.ejb.TransactionAttribute;
-import javax.ejb.TransactionAttributeType;
-import javax.enterprise.concurrent.ManagedExecutorService;
-import javax.jms.ConnectionFactory;
-import javax.naming.InitialContext;
-import javax.persistence.PostRemove;
-
-import org.hawkular.bus.common.ConnectionContextFactory;
-import org.hawkular.bus.common.Endpoint;
-import org.hawkular.bus.common.MessageProcessor;
-import org.hawkular.bus.common.consumer.ConsumerConnectionContext;
-import org.hawkular.cmdgw.ws.Constants;
-import org.hawkular.cmdgw.ws.MsgLogger;
-import org.hawkular.cmdgw.ws.mdb.AddDatasourceListener;
-import org.hawkular.cmdgw.ws.mdb.AddJdbcDriverListener;
-import org.hawkular.cmdgw.ws.mdb.DeployApplicationListener;
-import org.hawkular.cmdgw.ws.mdb.ExecuteOperationListener;
-import org.hawkular.cmdgw.ws.mdb.ExportJdrListener;
-
-@Startup
-@Singleton
-@TransactionAttribute(TransactionAttributeType.NOT_SUPPORTED)
-public class FeedListenerGenerator {
-    @EJB
-    private ConnectedFeeds connectedFeeds;
-
-    @Resource(mappedName = Constants.CONNECTION_FACTORY_JNDI)
-    private ConnectionFactory connectionFactory;
-
-    private Map<String, ConnectionContextFactory> connContextFactories;
-    private Map<String, List<ConsumerConnectionContext>> consumerContexts;
-
-    @Resource
-    private ManagedExecutorService threadPoolService;
-
-    @PostConstruct
-    public void initialize() throws Exception {
-        if (this.connectionFactory == null) {
-            MsgLogger.LOG.warnf("Injection of ConnectionFactory is not working - looking it up explicitly");
-            InitialContext ctx = new InitialContext();
-            this.connectionFactory = (ConnectionFactory) ctx.lookup(Constants.CONNECTION_FACTORY_JNDI);
-        } else {
-            MsgLogger.LOG.warnf("Injection of ConnectionFactory works - you can remove the hack");
-        }
-
-        connContextFactories = new HashMap<>();
-        consumerContexts = new HashMap<>();
-    }
-
-    @PostRemove
-    public void shutdown() throws Exception {
-        if (connContextFactories != null) {
-            for (String feedId : this.connContextFactories.keySet()) {
-                removeListeners(feedId);
-            }
-        }
-    }
-
-    /**
-     * @return the connection factory this object will use when connecting to the messaging system.
-     */
-    public ConnectionFactory getConnectionFactory() {
-        return this.connectionFactory;
-    }
-
-    public void addListeners(String feedId) throws Exception {
```

Comment:
If you examine how the old code did it, on each @OnOpen of the websocket, we added the listeners (that had message selectors appropriately set so the listeners would only listen for messages for the particular feed Id/ UI id) and on the @OnClose, we removed them.